src/test/java/uk/gov/hmcts/sptribs/idam/IdamServiceTest.java

```diff
@@ -91,6 +94,22 @@ public void shouldThrowFeignUnauthorizedExceptionWhenSystemUpdateUserCredentials
             .hasMessageContaining("Failed to retrieve Idam user");
     }
 
+    @Test
+    void shouldRetrieveUserDetails() {
+        // Given
+        String bearerToken = TEST_SERVICE_AUTH_TOKEN;
+        UserDetails userDetails = userDetails();
+
+        // When
+        when(idamClient.getUserDetails(bearerToken)).thenReturn(userDetails());
+
+        User result = idamService.retrieveUser(bearerToken);
+
+        //Then
+        assertEquals(userDetails, result.getUserDetails());
+        assertEquals(bearerToken, result.getAuthToken());
+    }
+
     private void setSystemUserCredentials() {
         ReflectionTestUtils.setField(idamService, "systemUpdateUserName", TEST_SYSTEM_UPDATE_USER_EMAIL);
```

Comment:
File - JudicialService.java
1. Mark all the private class variables as file
2. Move all the public methods to the top of the class and all private methods to the bottom of the class.